packages/flame_3d/lib/src/resources/light/light.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:flame_3d/game.dart';
+import 'package:flame_3d/resources.dart';
+
+/// {@template light}
+/// A [Resource] that represents a light source that is positioned in the scene
+/// and changes how other objects are rendered.
+///
+/// This class isn't a true resource, it does not upload it self to the GPU.
+/// Instead, it is used to modify how other resources are uploaded.
+///
+/// {@endtemplate}
+class Light extends Resource<void> {
+  final Transform3D transform;
+  final LightSource source;
+
+  /// {@macro light}
+  Light({
+    required this.transform,
+    required this.source,
+  }) : super(null);
+
+  void apply(Shader shader) {
+    shader.setVector3('Light.position', transform.position);
+    // apply additional parameters
+    source.apply(shader);
+  }
+
+  static UniformSlot shaderSlot = UniformSlot.value('Light', {'position'});
```

Comment:
The additional parameters would only work if they are part of the shader slot right? Shouldn't this be determined by the resource of a light then?

Reply:
I am not sure I understand this comment, I did extract it to the Light resource exactly because this class is responsible for defining and binding these values (right now only position but more in the future)

Reply:
discussed offline - we can address this when we add custom lights